src/cdomains/valueDomain.ml

```diff
@@ -1239,30 +1239,33 @@ struct
               in
               let offset = offs_to_offset offs in
 
-              let cast_to_void_ptr e = 
+              let cast_to_void_ptr e =
                 Cilfacade.mkCast ~e ~newt:(TPtr (TVoid [], []))
               in
               let i =
                 if InvariantCil.(not (exp_contains_tmp c_exp) && exp_is_in_scope scope c_exp && not (var_is_tmp vi) && var_is_in_scope scope vi && not (var_is_heap vi)) then
                   let addr_exp = AddrOf (Var vi, offset) in (* AddrOf or Lval? *)
                   let addr_exp, c_exp = if typeSig (typeOf addr_exp) <> typeSig (typeOf c_exp) then
                       cast_to_void_ptr addr_exp, cast_to_void_ptr c_exp
-                    else 
+                    else
                       addr_exp, c_exp
-                  in 
+                  in
                   Invariant.of_exp Cil.(BinOp (Eq, c_exp, addr_exp, intType))
                 else
                   Invariant.none
               in
               let i_deref =
+                (* Avoid dereferencing into functions, mutexes, ..., which will lead to errors as they are not added to the hash table *)
+                let noderef t = try isFunctionType t || Cilfacade.is_ignorable_type t with Not_found -> false in
```

Comment:
Where is this `Not_found` even coming from? It's a boolean function so it might as well handle that itself instead of requiring an extra try at every use.

Also, I think it should be just using `ValueDomain.is_immediate_type`.